### PR TITLE
Fall back to using ADC when setting up GCE PD CSI driver in CL2

### PR DESCRIPTION
**Pull request by tosi3k**
So far, it wasn't possible to set up GCE PD CSI driver in CL2 without providing service account credentials to the driver's controller.

This PR adds an option to reuse Application Default Credentials if `GOOGLE_APPLICATION_CREDENTIALS` env var is not set during CL2 execution.

/assign @jkaniuk

### CODE REVIEW

**Comment by tosi3k** [comment]
/hold

I want to test it end-to-end before it merges.

File: clusterloader2/run-e2e.sh [hunk 1]
```diff
@@ -29,14 +29,27 @@ export AZUREDISK_CSI_DRIVER_VERSION="v1.15.0"
...
 if [[ "${DEPLOY_GCI_DRIVER:-false}" == "true" ]]; then
    if [[ -z "${E2E_GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
-      echo "Env var E2E_GOOGLE_APPLICATION_CREDENTIALS must be set to deploy driver"
-      exit 1
+      # Running yq to patch GCE PD CSI Driver manifests so that it runs using
+      # Application Default Credentials instead of creating service account
+      # keys, hence avoiding security risks. See
+      # https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/610
+      # for more details.
+      yq eval -i '
```
Comment (jkaniuk) [inline clusterloader2/run-e2e.sh:40]: Do we have any guarantees that there is yq? What if there is not?

File: clusterloader2/run-e2e.sh [hunk 2]
```diff
@@ -29,14 +29,27 @@ export AZUREDISK_CSI_DRIVER_VERSION="v1.15.0"
...
+         with(select(.metadata.name == "csi-gce-pd-controller" and .kind == "Deployment").spec.template.spec;
+            del(.volumes[] | select(.name == "cloud-sa-volume")) |
+            with(.containers[] | select(.name == "gce-pd-driver");
+               del(.env[] | select(.name == "GOOGLE_APPLICATION_CREDENTIALS")) |
+               del(.volumeMounts[] | select(.name == "cloud-sa-volume"))
+            )
+         )' "${CLUSTERLOADER_ROOT}"/drivers/gcp-csi-driver-stable.yaml
```
Comment (jkaniuk) [inline clusterloader2/run-e2e.sh:47]: It should create a copy, not modify in place. In subsequent run, CL2 would behave differently with in-place modification.

File: clusterloader2/run-e2e.sh (hunk 1 above)
Comment (tosi3k) [reply clusterloader2/run-e2e.sh]: There is, if one runs CL2 from `kubekins-e2e`: https://github.com/kubernetes/test-infra/blob/ed307a9f6b5fe93c4004e4a2d803fa791c9318c0/images/kubekins-e2e/Dockerfile#L73-L76

Added a precautionary `if` statement with instruction on how to mitigate this.

File: clusterloader2/run-e2e.sh (hunk 2 above)
Comment (tosi3k) [reply clusterloader2/run-e2e.sh]: Good point; done.

File: clusterloader2/run-e2e.sh [hunk 3]
```diff
@@ -29,14 +29,34 @@ export AZUREDISK_CSI_DRIVER_VERSION="v1.15.0"
 export AZUREDISK_CSI_DRIVER_INSTALL_URL="https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/${AZUREDISK_CSI_DRIVER_VERSION}/deploy/install-driver.sh"
 export WINDOWS_USE_HOST_PROCESS_CONTAINERS=true
 
-# Deploy the GCP PD CSI Driver if required
+# Deploy the GCE PD CSI Driver if required
 if [[ "${DEPLOY_GCI_DRIVER:-false}" == "true" ]]; then
    if [[ -z "${E2E_GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
-      echo "Env var E2E_GOOGLE_APPLICATION_CREDENTIALS must be set to deploy driver"
```
Comment (jkaniuk) [inline clusterloader2/run-e2e.sh:35]: We've lost this old message about application credentials. Is that not relevant anymore?

Comment (jkaniuk) [reply clusterloader2/run-e2e.sh:35]: Also - is there any way to distinguish between the case when user forgot credentials and when forgot yq? Current message could be quite confusing.

File: clusterloader2/run-e2e.sh [hunk 4]
```diff
@@ -29,14 +29,34 @@ export AZUREDISK_CSI_DRIVER_VERSION="v1.15.0"
 export AZUREDISK_CSI_DRIVER_INSTALL_URL="https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/${AZUREDISK_CSI_DRIVER_VERSION}/deploy/install-driver.sh"
 export WINDOWS_USE_HOST_PROCESS_CONTAINERS=true
 
-# Deploy the GCP PD CSI Driver if required
+# Deploy the GCE PD CSI Driver if required
 if [[ "${DEPLOY_GCI_DRIVER:-false}" == "true" ]]; then
    if [[ -z "${E2E_GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
```
Comment (jkaniuk) [inline clusterloader2/run-e2e.sh:34]: I would reverse E2E_GOOGLE_APPLICATION_CREDENTIALS check - first case with credentials, then longer case of yq

File: clusterloader2/run-e2e.sh (hunk 3 above)
Comment (tosi3k) [reply clusterloader2/run-e2e.sh:35]: Added informative message about the lack of credentials.

**Pull request merged**